### PR TITLE
Json schema

### LINKED ISSUE #18: Implement JSON-SCHEMA Model
DOD:

- [x] Implement JSON-SCHEMA model defined in other ticket
- [x] Implement Tests for this model
- [ ] Write new Tickets expanding the SCHEMA

### PR DESCRIPTION
**Pull request by jp06upel**
Closes #18

### CODE REVIEW

File: doc/topics/models/Components.png
Comment (CodeOfRob) [inline doc/topics/models/Components.png:1]: connection missing between `property:Integer/Number` and `Property`?

File: src/main/java/org/jqassistant/plugin/openapi/api/model/ComponentsDescriptor.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import com.buschmais.xo.neo4j.api.annotation.Label;
 import com.buschmais.xo.neo4j.api.annotation.Relation;
+import org.jqassistant.plugin.openapi.api.model.jsonschema.SchemaDescriptor;
```
Comment (CodeOfRob) [inline src/main/java/org/jqassistant/plugin/openapi/api/model/ComponentsDescriptor.java:5]: import necessary?

File: doc/topics/models/Components.png
Comment (jp06upel) [reply doc/topics/models/Components.png:1]: done

**Review by CodeOfRob — APPROVED** [review]
approved but at this extent **more than one review suggested**

File: src/test/java/org/jqassistant/plugin/openapi/parser/JsonSchemaTest.java [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        assertThat(contract).isNotNull();
+    }
+
+    @AfterEach
+    void closeTransaction(){
+        store.commitTransaction();
+    }
+
+    @Test
+    void BasicTest(){
```
Comment (UzeyirMammadov) [inline src/test/java/org/jqassistant/plugin/openapi/parser/JsonSchemaTest.java:40]: I would suggest coming up with a more descriptive name for this test. If it fails it won't be clear what actually failed.

File: src/test/java/org/jqassistant/plugin/openapi/parser/JsonSchemaTest.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        ComponentsDescriptor componentsDescriptor = contract.getComponents();
+        assertThat(componentsDescriptor).isNotNull();
+
+        assertThat(componentsDescriptor.getSchemas()).isNotNull();
+        assertThat(componentsDescriptor.getSchemas()).hasSizeGreaterThan(1);
+    }
+
+    @Test
+    void integerSchemaBasicTest(){
```
Comment (UzeyirMammadov) [inline src/test/java/org/jqassistant/plugin/openapi/parser/JsonSchemaTest.java:49]: same here

File: src/test/java/org/jqassistant/plugin/openapi/parser/JsonSchemaTest.java [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+        PropertyDescriptor propertyDescriptor = schemaDescriptor.getObject();
+
+        assertThat(propertyDescriptor).isNotNull();
+        assertThat(propertyDescriptor.getName()).isEqualTo("IntegerSchema");
+        assertThat(propertyDescriptor).isInstanceOf(IntegerPropertyDescriptor.class);
+    }
+
+    @Test
+    void schemaWithObjectTest(){
```
Comment (UzeyirMammadov) [inline src/test/java/org/jqassistant/plugin/openapi/parser/JsonSchemaTest.java:63]: Some names have a lowercase first letter, some have an uppercase first letter. Would suggest to be consistent.

**Review by UzeyirMammadov — COMMENTED** [review]
Reviw done. Please check my comments.

Nice work! 👍

File: src/test/java/org/jqassistant/plugin/openapi/parser/JsonSchemaTest.java (hunk 2 above)
Comment (jp06upel) [reply src/test/java/org/jqassistant/plugin/openapi/parser/JsonSchemaTest.java]: renamed

File: src/test/java/org/jqassistant/plugin/openapi/parser/JsonSchemaTest.java (hunk 4 above)
Comment (jp06upel) [reply src/test/java/org/jqassistant/plugin/openapi/parser/JsonSchemaTest.java:63]: renamed so all functions start with lower case

**Review by CodeOfRob — APPROVED** [review]

**Review by UzeyirMammadov — APPROVED** [review]

**Pull request merged**